pyproject.toml

```diff
@@ -65,20 +65,17 @@ classifiers = [
     "Topic :: Scientific/Engineering :: Bio-Informatics",
 ]
 dependencies = [
+    "anysqlite", 
+    "hishel[sqlite]",
```

Comment:
I think we still want to keep "caching" dependencies as optional?

Reply:
Oh yeah actually I should have mentioned in our discussion this may be a lot harder to maintain separating the caching dependencies as separate. In order to maintain caching for both sync and async with this new packaging there is a far stronger coupling between the client instance and the cache. See the method `_build_http_client` (similar between `client/base.py` and `client/asynchronous.py`. 

```
   99     def _build_http_client(self, cache_db: Union[str, Path] = None) -> None:
  100         """
  101         Builds the client instance for usage through the lifetime
  102         of the biothings_client
  103         """
  104         if not self.client_setup_completed:
  105             if cache_db is None:
  106                 cache_db = self._default_cache_file
  107             cache_db = Path(cache_db).resolve().absolute()
  108 
  109             self.cache_storage = BiothingsClientSqlite3Cache()
  110             self.cache_storage.setup_database_connection(cache_db)
  111             cache_transport = hishel.CacheTransport(transport=httpx.HTTPTransport(), storage=self.cache_storage)
  112             cache_controller = hishel.Controller(cacheable_methods=["GET", "POST"])
  113             self.http_client = hishel.CacheClient(
  114                 controller=cache_controller, transport=cache_transport, storage=self.cache_storage
  115             )
  116             self.client_setup_completed = True
  ```
We build the caching storage instance, the HTTP transport, the HTTP controller, and the client. With `pandas` that package is fairly heavy so I'm fine providing a barrier for that. However in this case the caching libraries are fairly small and light-weight. Plus the user must enable it through the `set_caching` method. I would worry about modifying the state of the client in the `set_caching` and `stop_caching` methods if we want to dynamically change the transport and storage instances. If you have an idea about how we could cleanly decouple it then I think we could try that but most of the examples I see of it online have this coupling. I used the following [client](https://github.com/parkerhancock/patent_client) library as a reference while building this if you want to see another client type library leveraging `httpx` and `hishel`

Reply:
so if caching is turned off (that's the default), will there be a local sqlite file created, but won't be used?

Reply:
Correct that is the current behavior. However because we override the cache storage we can likely control that behavior if you would prefer to only create it when we enable caching

Reply:
yeah, I would expect no cache file is created until it's explicitly enabled.